### PR TITLE
Finish TokenMarker RE, implement in Devise a Stratagem

### PR DESCRIPTION
**Pull request by stwlam**

### CODE REVIEW

**Comment by CarlosFdez** [comment]
Is it possible for strategic strike to work if you remove the roll option? Unfortunately the effect is set to auto remove, which means its gone when damage occurs.

File: src/module/rules/rule-element/token-marker/rule-element.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+                : await new MarkTargetPrompt({ prompt: null, requirements: null }).resolveTarget());
+        if (!(token instanceof TokenDocumentPF2e)) {
+            // No token was targeted: abort creating item
+            pendingItems.splice(pendingItems.indexOf(itemSource), 1);
+            return;
+        }
+
+        this.#checkRuleSource(ruleSource);
+        this.uuid = ruleSource.uuid = token.uuid;
+        itemSource.name = `${itemSource.name} (${token.name})`;
```
Comment (In3luki) [inline src/module/rules/rule-element/token-marker/rule-element.ts:38]: Does `token.name` respect the `tokenSetsNameVisibility` setting?

Comment (stwlam) [reply src/module/rules/rule-element/token-marker/rule-element.ts]: I removed it and will follow up with some visual indication on tokens.

File: src/module/rules/rule-element/token-mark/rule-element.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+            (game.user.targets.size === 1
+                ? Array.from(game.user.targets)[0]!.document
+                : await new MarkTargetPrompt({ prompt: null, requirements: null }).resolveTarget());
+        if (!(token instanceof TokenDocumentPF2e)) {
+            // No token was targeted: abort creating item
+            pendingItems.splice(pendingItems.indexOf(itemSource), 1);
+            return;
+        }
+
+        this.#checkRuleSource(ruleSource);
```
Comment (CarlosFdez) [inline src/module/rules/rule-element/token-mark/rule-element.ts:41]: Why is this check necessary? 👀

Comment (stwlam) [reply src/module/rules/rule-element/token-mark/rule-element.ts:41]: Because otherwise the object has no `uuid` property.

Comment (CarlosFdez) [reply src/module/rules/rule-element/token-mark/rule-element.ts:41]: That can cause a rule source to not have the key set? I'd assume this wouldn''t be executed if the key wasn't TokenMark already.

Comment (stwlam) [reply src/module/rules/rule-element/token-mark/rule-element.ts:41]: It can't do that, no.

**Pull request merged**